cmd/kubeadm/app/util/config/common.go

```diff
@@ -91,7 +91,7 @@ func validateSupportedVersion(gv schema.GroupVersion, allowDeprecated, allowExpe
 	}
 
 	if _, present := deprecatedAPIVersions[gvString]; present && !allowDeprecated {
-		klog.Warningf("your configuration file uses a deprecated API spec: %q. Please use 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.", gv)
+		klog.Warningf("your configuration file uses a deprecated API spec: %q. Please use 'kubeadm config migrate --old-config old.yaml --new-config new.yaml', which will write the new, similar spec using a newer API version.", gv.String())
```

Comment:
%q automatically calls the String method, no?
the change is fine, but wondering - did you get a linter warning / error?

Reply:
> %q automatically calls the String method, no?

Maye I missed something, but looks like to me %q is mostly want to escape the character in the quote.

> Integer
> %q	a single-quoted character literal safely escaped with Go syntax.
>
> String and slice of bytes
> %q	a double-quoted string safely escaped with Go syntax

I haven't notice any warn/error if we don't change this, but `gv.String()` looks more explicit here that it's will be going to be a string.

Reply:
ok